### ISSUE TITLE
Enable fail2ban for postfix

### ISSUE THREAD

**Issue by jchristgit** [python-discord/infra#433]
Spammers, scanners and associates of Connor McFarlane have a neverending desire to scan the internet to look for open relays.

I propose we stop wasting our resources (and log line space) by enabling the fail2ban jail for postfxi.

I have this on my personal installation, which is probably as brutal as it gets, but works well:

```ini
[postfix]
mode = aggressive
enabled = true
bantime = 1d

[postfix-submission]
filter = postfix[mode=aggressive]
port = submission
enabled = true
bantime = 1d

[postfix-errors]
filter = postfix[mode=errors]
enabled = true
maxretry = 2
bantime = 5y
```

Note the differentiation between regular postfix and submission port (authenticated users sending outbound mail). The submission section is probably useless for us, but the other one might be useful.

**Comment by jb3** [comment]
The commit to fix this (55df090) errored when deployed to production, it has been reverted in 8a2e9d6

```
Aug 06 20:00:34 lovelace.box.pydis.wtf fail2ban-server[2163850]: 2024-08-06 20:00:34,190 fail2ban                [2163850]: ERROR   Failed during configuration: Have not found any log file for postfix jail
Aug 06 20:00:34 lovelace.box.pydis.wtf fail2ban-server[2163850]: 2024-08-06 20:00:34,192 fail2ban                [2163850]: ERROR   Async configuration of server failed
```

**Comment by jb3** [comment]
This is done.
